### PR TITLE
Nitro in backend

### PR DESCRIPTION
**Pull request by Wenzel**
Moved the `Nitro` object initialization into the `Backend`.

The end goal is to modify the callback API to make it look like this
~~~
def enter_NtOpenFile(backend, syscall):
~~~
including a new parameter, the `backend` itself.
This will allow the callback to query information of the `backend`, and even more, give instructions to `nitro`, like getting the register or setting them.

At the end, we allow the developer to change the syscall parameters on the fly (if they are stored in registers)

### CODE REVIEW

File: nitro/backend.py [hunk 1]
```diff
@@ -76,38 +77,46 @@ class Backend:
...
+            self.sdt = None
+            self.load_symbols()
+            # run libvmi helper subprocess
+            self.libvmi = Libvmi(domain.name())
+            self.processes = {}
+            self.hooks = {
+                SyscallDirection.enter: {},
+                SyscallDirection.exit: {}
+            }
+            self.stats = {
```
Comment (noxdafox) [inline nitro/backend.py]:
```
self.stats = defaultdict(int) 
```
Will provide all keys initialized to 0.

File: nitro/backend.py [hunk 2]
```diff
@@ -76,38 +77,46 @@ class Backend:
...
+            }
 
     def __enter__(self):
         return self
 
     def __exit__(self, exc_type, exc_value, traceback):
-        self.stop()
+        if self.analyze:
+            self.stop()
+        self.nitro.stop()
```
Comment (noxdafox) [inline nitro/backend.py]: What if `nitro.stop()` gets explicitly called by the user before exiting from the context manager?

File: nitro/backend.py [hunk 3]
```diff
@@ -76,38 +77,46 @@ class Backend:
         'processes',
         'hooks',
         'stats',
+        'nitro',
+        'analyze'
     )
 
-    def __init__(self, domain):
+    def __init__(self, domain, analyze=False):
```
Comment (noxdafox) [inline nitro/backend.py]: Couple of notes here:
1) the Backend class has far too many attributes. Consider using composition to group them in objects.
2) Several of these attributes get initialized only if a condition `analyze` is `True`. This is very error prone as you cannot track in the below methods which attribute exists or not. Always initialize all attributes to `None`/{}/[]/''.

File: nitro/backend.py (hunk 2 above)
Comment (Wenzel) [inline nitro/backend.py]: Calling `nitro.stop()` multiple times will not cause any issue if Nitro is already stopped before.

File: nitro/backend.py (hunk 3 above)
Comment (Wenzel) [inline nitro/backend.py]: This is not the final design but a temporary one.
The `backend` will have to be renamed `engine` and splitted into multiple objects.

**Pull request merged**